src/devices/Servo/ServoMotor.cs

```diff
@@ -0,0 +1,196 @@
+
+        double currentPulseWidth = 0;
+
+        private float angle;
+        /// <summary>
+        /// Servo motor definition.
+        /// </summary>
+        private ServoMotorDefinition definition;
+        /// <summary>
+        /// The duration per angle's degree.
+        /// </summary>
+        private double rangePerDegree;
+        private PwmController pwmController;
+
+        public ServoMotor(int pinNumber, int pwmChannel, ServoMotorDefinition definition)
+        {                       
+            this.definition = definition;
+            PulseFrequency = definition.Period / 1000.0;
+
+            UpdateRange();
+
+            servoPin = pinNumber;
+            this.pwmChannel = pwmChannel;
+            // try hardware PWM first
+            try{
+                pwmController = new PwmController();
+                pwmController.OpenChannel(pinNumber, pwmChannel);
+                Console.WriteLine("Using hardware PWM");
+
+            } catch
+            {
+                pwmController = new PwmController(new SoftPwm(true));
+                pwmController.OpenChannel(pinNumber, pwmChannel);
+                Console.WriteLine("Using software PWM");
+            }
+
+            pwmController.StartWriting(servoPin, pwmChannel, 1000 / PulseFrequency,  (1 - (PulseFrequency - currentPulseWidth)/PulseFrequency) * 100);         
+
+        }
+
+        /// <summary>
+        /// Rotate the servomotor with a specific pulse in microseconds
+        /// </summary>
+        /// <param name="duration">Pulse duration in microseconds</param>
+        public void SetPulse(uint duration)
+        {
+            currentPulseWidth = duration / 1000.0;
+            pwmController.ChangeDutyCycle(servoPin, pwmChannel, (1 - (PulseFrequency - currentPulseWidth)/PulseFrequency) * 100);
+        }
+        /// <summary>
+        /// Rotate the servomotor with a specific pulse in microseconds
+        /// </summary>
+        /// <param name="period">Period duration in microseconds</param>
+        /// <param name="duration">Pulse duration in microseconds</param>
+        /// <remarks>Changing the period will override the period definition for the motor. It is not recommended to do so.</remarks>
+        public void SetPulse(uint period, uint duration)
+        {
+            definition.Period = period;
+            PulseFrequency = period / 1000.0;
+            currentPulseWidth = duration / 1000.0;
+            pwmController.ChangeDutyCycle(servoPin, pwmChannel, (1 - (PulseFrequency - currentPulseWidth)/PulseFrequency) * 100);
+        }
+        /// <summary>
+        /// Is it moving? Return true if the servomotor is currently moving.
+        /// </summary>
+        public bool IsMoving
+        {
+            get
+            {
+                if (currentPulseWidth != 0)
+                    return true;
+                else
+                    return false;
+            }
+        }
+
+        /// <summary>
+        /// Current angle.
```

Comment:
I think this needs some comment about the units (degrees vs radians) and range. Optionally instead of clamping you could modulo. If you stay with clamping you can use Math.Clamp instead of ifs